tests/test_entrypoint.py

```diff
@@ -0,0 +1,239 @@
+    - CLI entrypoints
+    - GUI entrypoints
+    - miscellaneous entrypoints
+"""
+
+from setuptools_pyproject_migration import WritePyproject
+from setuptools.dist import Distribution
+
+
+# Entry point parsing
+
+
+def test_parse_entrypoint_happypath():
+    """
+    Test we can extract the entry point name and target from a string.
+    """
+    assert WritePyproject._parse_entrypoint("ep=project.module:target") == ("ep", "project.module:target")
+
+
+def test_parse_entrypoint_happypath_whitespace():
+    """
+    Test we strip extraneous whitespace from the strings received.
+    """
+    assert WritePyproject._parse_entrypoint("  ep  =  project.module:target  ") == ("ep", "project.module:target")
+
+
+def test_parse_entrypoint_missing_eq():
+    """
+    Test that a string lacking a '=' is refused.
+    """
+    try:
+        WritePyproject._parse_entrypoint("ep project.module:target")
+        assert False, "Should not have been accepted as valid"
+    except ValueError as e:
+        assert str(e).endswith(" is not of the form 'name = module:function'")
+
+
+# Wrapper function logic
+
+
+def test_generate_entrypoints_none():
+    """
+    Test we get an empty dict if entrypoints is None.
+    """
+    assert WritePyproject._generate_entrypoints(None) == {}
+
+
+def test_generate_entrypoints():
+    """
+    Test we get all entry points grouped by type.
+    """
+    assert WritePyproject._generate_entrypoints(
+        {
+            "console_scripts": [
+                "spanish-inquisition = montypython.unexpected:spanishinquisition",
+                "brian=montypython.naughtyboy:brian",
+            ],
+            "gui_scripts": [
+                "dead-parrot=montypython.sketch:petshop",
+                "shrubbery=montypython.holygrail:knightswhosayni",
+            ],
+            "project.plugins": [
+                "babysnatchers=montypython.somethingcompletelydifferent:babysnatchers",
+                "eels=montypython.somethingcompletelydifferent:eels",
+            ],
+        }
+    ) == {
+        "console_scripts": {
+            "spanish-inquisition": "montypython.unexpected:spanishinquisition",
+            "brian": "montypython.naughtyboy:brian",
+        },
+        "gui_scripts": {
+            "dead-parrot": "montypython.sketch:petshop",
+            "shrubbery": "montypython.holygrail:knightswhosayni",
+        },
+        "project.plugins": {
+            "babysnatchers": "montypython.somethingcompletelydifferent:babysnatchers",
+            "eels": "montypython.somethingcompletelydifferent:eels",
+        },
+    }
+
+
+# Test correct placement of the parsed endpoints
+
+
+def test_generate_noentrypoints():
+    """
+    Test distribution with no entry points generates no entry points
+    """
+    cmd = WritePyproject(
+        Distribution(
+            dict(
+                name="TestProject",
+                version="1.2.3",
+            )
+        )
+    )
+    project = cmd._generate()
```

Comment:
Minor thing, but in other tests I've been using `result` as the name of the data structure that gets returned from `_generate()`, and it might be handy to stay consistent on that. (I mean, `result` isn't a great name, and we could definitely change it, but the consistency is what matters.) It might throw someone off a bit to use `project` when that is also the name of a fixture that we use prominently in other tests.

Reply:
Indeed, I guess I was thinking about the object data type (`_generate()` returns a `Project`; and `project` here is an instance of it), but as we're using that for a fixture, perhaps the more generic `result` is better.